homeassistant/components/enphase_envoy/binary_sensor.py

```diff
@@ -124,7 +192,79 @@ def __init__(
             via_device=(DOMAIN, self.envoy_serial_num),
         )
 
+    @property
+    def is_on(self) -> bool:
+        """Return the state of the Encharge binary_sensor."""
+        encharge_inventory = self.data.encharge_inventory
+        assert encharge_inventory is not None
+        return self.entity_description.value_fn(encharge_inventory[self._serial_number])
+
+
+class EnvoyEnpowerBinarySensorEntity(EnvoyBaseBinarySensorEntity):
+    """Defines an Enpower binary_sensor entity."""
+
+    entity_description: EnvoyEnpowerBinarySensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: EnphaseUpdateCoordinator,
+        description: EnvoyEnpowerBinarySensorEntityDescription,
+    ) -> None:
+        """Init the Enpower base entity."""
+        super().__init__(coordinator, description)
+        enpower = self.data.enpower
+        assert enpower is not None
+        self._serial_number = enpower.serial_number
+        self._attr_unique_id = f"{self._serial_number}_{description.key}"
+        assert enpower is not None
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self._serial_number)},
+            manufacturer="Enphase",
+            model="Enpower",
+            name=f"Enpower {self._serial_number}",
+            sw_version=str(enpower.firmware_version),
+            via_device=(DOMAIN, self.envoy_serial_num),
+        )
+
+    @property
+    def is_on(self) -> bool:
+        """Return the state of the Enpower binary_sensor."""
+        enpower = self.data.enpower
+        assert enpower is not None
+        return self.entity_description.value_fn(enpower)
+
+
+class EnvoyRelayBinarySensorEntity(
+    CoordinatorEntity[EnphaseUpdateCoordinator], BinarySensorEntity
+):
```

Comment:
```suggestion
class EnvoyRelayBinarySensorEntity(EnvoyBaseBinarySensorEntity):
```

Reply:
Will this work since the relay status sensors don't have an entity description?

Reply:
No it won't. I missed that there is no description

Reply:
But you could add one

```
 
+RELAY_DESCRIPTION = BinarySensorEntityDescription(
+    has_entity_name=True,
+    icon="mdi:power-plug",
+)
```

```
         entities.extend(
-            EnvoyRelayBinarySensorEntity(coordinator, relay)
+            EnvoyRelayBinarySensorEntity(coordinator, RELAY_DESCRIPTION, relay)
             for relay in envoy_data.dry_contact_status
         )
```

Reply:
You could also write it like this
```python
class EnvoyRelayBinarySensorEntity(EnvoyBaseBinarySensorEntity):
    """Defines an Enpower dry contact binary_sensor entity."""

    def __init__(
        self,
        coordinator: EnphaseUpdateCoordinator,
        relay: str,
    ) -> None:
        """Init the Enpower base entity."""
        description = replace(
            RELAY_DESCRIPTION,
            name=f"{self.data.dry_contact_settings[self.relay.id].load_name} Relay"
        )
        super().__init__(coordinator, description)
        enpower = self.data.enpower
        assert enpower is not None
        self.relay = self.data.dry_contact_status[relay]
        self._serial_number = enpower.serial_number
        self._attr_unique_id = f"{self._serial_number}_relay_{self.relay.id}"
        self._attr_device_info = DeviceInfo(
            identifiers={(DOMAIN, self._serial_number)},
            manufacturer="Enphase",
            model="Enpower",
            name=f"Enpower {self._serial_number}",
            sw_version=str(enpower.firmware_version),
            via_device=(DOMAIN, self.envoy_serial_num),
        )

    @property
    def is_on(self) -> bool:
        """Return the state of the Enpower binary_sensor."""
        return self.relay.status == DryContactStatus.CLOSED
```